src/import_edf.jl

```diff
@@ -208,81 +302,264 @@ function groupby(f, list)
-if channels with the same signal kind/type have different sample rates or
-physical units.
+plan_edf_to_onda_samples(signal::EDF.Signal, s; kwargs...) = plan_edf_to_onda_samples(_named_tuple(signal.header), s; kwargs...)
+plan_edf_to_onda_samples(header::EDF.SignalHeader, s; kwargs...) = plan_edf_to_onda_samples(_named_tuple(header), s; kwargs...)
 
-`(label, transducer_type)` pairs will be transformed into labels by `preprocess_labels`
-(default preprocessor returns the original label). An alternate to `STANDARD_UNITS`
-for mapping different spellings of units to their canonical unit name can be passed in
-as `unit_alternatives`.
+"""
+    plan_edf_to_onda_samples(header, seconds_per_record; labels=STANDARD_LABELS,
+                             units=STANDARD_UNITS, preprocess_labels=(l,t) -> l)
+    plan_edf_to_onda_samples(signal::EDF.Signal, args...; kwargs...)
+
+Formulate a plan for converting an EDF signal into Onda format.  This returns a
+Tables.jl row with all the columns from the signal header, plus additional
+columns for the `Onda.SamplesInfo` for this signal, and the `seconds_per_record`
+that is passed in here.
+
+If no labels match, then the `channel` and `kind` columns are `missing`; the
+behavior of other `SamplesInfo` columns is undefined; they are currently set to
+missing but that may change in future versions.
+
+Any errors that are thrown in the process will be stored as `SampleInfoError`s
+in the `error` column.
+
+## Matching EDF label to Onda labels
+
+The `labels` keyword argument determines how Onda `channel` and signal `kind`
+are extracted from the EDF label.
+
+Labels are specified as an iterable of `signal_names => channel_names` pairs.
+`signal_names` should be an iterable of signal names, the first of which is the
+canonical name used as the Onda `kind`.  Each element of `channel_names` gives
+the specification for one channel, which can either be a string, or a
+`canonical_name => alternates` pair.  Occurences of `alternates` will be
+replaces with `canonical_name` in the generated channel label.
+
+Matching is determined _solely_ by the channel names.  When matching, the signal
+names are only used to remove signal names occuring as prefixes (e.g., "[ECG]
+AVL") before matching channel names.  See [`match_edf_label`](@ref) for details,
+and see `OndaEDF.STANDARD_LABELS` for the default labels.
+
+As an example, here is (a subset of) the default labels for ECG signals:
+
+```julia
+["ecg", "ekg"] => ["i" => ["1"], "ii" => ["2"], "iii" => ["3"],
+                   "avl"=> ["ecgl", "ekgl", "ecg", "ekg", "l"], 
+                   "avr"=> ["ekgr", "ecgr", "r"], ...]
+```
+
+Matching is done in the order that `labels` iterates pairs, and will stop at the
+first match, with no warning if signals are ambiguous (although this may change
+in a future version)
+"""
+function plan_edf_to_onda_samples(header,
+                                  seconds_per_record=_get(header,
```

Comment:
Is it ever the case that you'd want to call this function with some arbitrary value of `seconds_per_record` that doesn't necessarily match what was in the file header? If not, you can move your `ismissing` check below the `row` definition and use `ismissing(_get(row, :seconds_per_record))`. It will have it if `header` has it.